src/main/java/org/opensearch/neuralsearch/processor/DocumentChunkingProcessor.java

```diff
@@ -0,0 +1,333 @@
+    private Map<String, Object> chunkerParameters;
+
+    private final Map<String, Object> fieldMap;
+
+    private final ClusterService clusterService;
+
+    private final IndicesService indicesService;
+
+    private final AnalysisRegistry analysisRegistry;
+
+    private final Environment environment;
+
+    public DocumentChunkingProcessor(
+        String tag,
+        String description,
+        Map<String, Object> fieldMap,
+        Map<String, Object> algorithmMap,
+        Environment environment,
+        ClusterService clusterService,
+        IndicesService indicesService,
+        AnalysisRegistry analysisRegistry
+    ) {
+        super(tag, description);
+        validateAndParseAlgorithmMap(algorithmMap);
+        this.fieldMap = fieldMap;
+        this.environment = environment;
+        this.clusterService = clusterService;
+        this.indicesService = indicesService;
+        this.analysisRegistry = analysisRegistry;
+    }
+
+    public String getType() {
+        return TYPE;
+    }
+
+    @SuppressWarnings("unchecked")
+    private void validateAndParseAlgorithmMap(Map<String, Object> algorithmMap) {
+        if (algorithmMap.size() != 1) {
+            throw new IllegalArgumentException(
+                "Unable to create the processor as [" + ALGORITHM_FIELD + "] must contain and only contain 1 algorithm"
+            );
+        }
+
+        for (Map.Entry<String, Object> algorithmEntry : algorithmMap.entrySet()) {
+            String algorithmKey = algorithmEntry.getKey();
+            Object algorithmValue = algorithmEntry.getValue();
+            if (!supportedChunkers.contains(algorithmKey)) {
+                throw new IllegalArgumentException(
+                    "Unable to create the processor as chunker algorithm ["
+                        + algorithmKey
+                        + "] is not supported. Supported chunkers types are ["
+                        + FIXED_LENGTH_ALGORITHM
+                        + ", "
+                        + DELIMITER_ALGORITHM
+                        + "]"
+                );
+            }
+            if (!(algorithmValue instanceof Map)) {
+                throw new IllegalArgumentException(
+                    "Unable to create the processor as [" + algorithmKey + "] parameters cannot be cast to [" + Map.class.getName() + "]"
+                );
+            }
+            FieldChunker chunker = ChunkerFactory.create(algorithmKey, analysisRegistry);
+            chunker.validateParameters((Map<String, Object>) algorithmValue);
+            this.chunkerType = algorithmKey;
+            this.chunkerParameters = (Map<String, Object>) algorithmValue;
+            chunker.validateParameters(chunkerParameters);
+            if (((Map<String, Object>) algorithmValue).containsKey(MAX_CHUNK_LIMIT_FIELD)) {
+                int max_chunk_limit = ((Number) ((Map<String, Object>) algorithmValue).get(MAX_CHUNK_LIMIT_FIELD)).intValue();
+                if (max_chunk_limit <= 0) {
```

Comment:
Default value is -1, and when user configure -1, we will throw this exception? please change 0 to -1 here and change the error message accordingly.

Reply:
The `max_chunk_limit` parameter is not enabled by default. How about change this if statement to
`if (max_chunk_limit <= 0 && max_chunk_limit != DEFAULT_MAX_CHUNK_LIMIT)`

Reply:
Isn't that `if (max_chunk_limit <= 0 && max_chunk_limit != DEFAULT_MAX_CHUNK_LIMIT)` is equals to `if (max_chunk_limit < DEFAULT_MAX_CHUNK_LIMIT)`?

Reply:
I think `max_chunk_limit = 0` is also invalid